Investigate flapping resolv.conf unit test

The integration test around validating the resolv.conf propagates to the apps appears to be a flapper:

http://ci.kurma.io/project.html?projectId=Kurma_IntegrationTests&testNameId=514927914447376594&tab=testDetails

All local testing passed without issue, as well as the initial runs on the PR. But once merged to master, saw the first failures. The PR branch was fully up to date with master.

Need to investigate further. It doesn't appear to be erroring about the file being there, rather it seems to just be the resolv.conf is blank.

It would be good to watch the unit tests as the scheduled runs add more to them to perhaps narrow it down:

http://ci.kurma.io/project.html?projectId=Kurma&testNameId=-9181810187844877476&tab=testDetails
http://ci.kurma.io/project.html?projectId=Kurma&testNameId=5919699581752510096&tab=testDetails

It might be that upon exit of with entering a container, the remaining output may not be getting flushed. From the test long on a successful run:

```
I0331 19:18:24.254812 #2620] ==> /checkout/src/github.com/apcera/kurma/testing/integration/lib/kurma/../../../../bin/kurma-cli enter c77c7156-45d3-4340-9998-42c69498ac5b busybox
I0331 19:18:24.301195 #2620] STDOUT: cat /etc/resolv.conf
exit
/ # cat /etc/resolv.conf
search us-west-2.compute.internal
nameserver 172.20.0.2
/ # exit

I0331 19:18:24.301809 #2620] ==> /checkout/src/github.com/apcera/kurma/testing/integration/lib/kurma/../../../../bin/kurma-cli stop c77c7156-45d3-4340-9998-42c69498ac5b
I0331 19:18:24.360372 #2620] STDOUT: Destroyed pod c77c7156-45d3-4340-9998-42c69498ac5b

E0331 19:18:24.361608 #2620] --- PASS: Container networking should create a resolv.conf for containers
```

And a failed run:

```
I0331 20:33:35.643572 #2622] ==> /checkout/src/github.com/apcera/kurma/testing/integration/lib/kurma/../../../../bin/kurma-cli enter 02f4e14e-d3ee-43c5-9ce3-3e924f5e59df busybox
I0331 20:33:35.696157 #2622] STDOUT: cat /etc/resolv.conf
exit

E0331 20:33:35.697411 #2622] --- FAIL: Container networking should create a resolv.conf for containers
```

Both are from the Fedora 22 runs for consistency.

In the successful one, the commands that are inputted are echo'd back, however in the failed run, the commands are not echo'd at all.

--- a/stager/container/run/run.go
+++ b/stager/container/run/run.go
@@ -7,6 +7,7 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"sync"
 
 	"github.com/apcera/kurma/schema"
 	"github.com/opencontainers/runc/libcontainer"
@@ -31,6 +32,10 @@ func Run() error {
 		return err
 	}
 
+	// Allocate a wait group which is primarily used when a tty is requested, to
+	// ensure all content is written before returning.
+	wg := sync.WaitGroup{}
+
 	// Setup the process
 	workingDirectory := app.WorkingDirectory
 	if workingDirectory == "" {
@@ -54,7 +59,11 @@ func Run() error {
 		if err != nil {
 			return err
 		}
-		go io.Copy(os.Stdout, console)
+		wg.Add(1)
+		go func() {
+			io.Copy(os.Stdout, console)
+			wg.Done()
+		}()
 		go io.Copy(console, os.Stdin)
 	}
 
@@ -63,5 +72,9 @@ func Run() error {
 		return err
 	}
 	process.Wait()
+
+	// Wait for other routines to finish up and flush output
+	wg.Wait()
+
 	return nil
 }

--- a/testing/integration/spec/create_spec.rb
+++ b/testing/integration/spec/create_spec.rb
@@ -35,4 +35,17 @@ RSpec.describe "CLI create" do
     resp = api.list_pods
     expect(resp["pods"].size).to eq(initial_pods_count)
   end
+
+  it "should enter a container and run a command" do
+    output = cli.run!("create docker://busybox --name busybox --net=host /bin/sleep 60")
+    uuid = output.scan(/Launched pod ([\w-]+)/).flatten.first
+    expect(uuid).not_to be_nil
+
+    output = cli.run!("enter #{uuid} busybox", "whoami", "exit")
+    output.gsub!("\r", "") # trim carriage returns
+    expect(output).to match("whoami\nroot")
+
+    output = cli.run!("stop #{uuid}")
+    expect(output).to include("Destroyed pod")
+  end
 end

